src/options.rs

```diff
@@ -667,3 +693,26 @@ impl ReconnectDelayCallback {
         self.0(reconnects)
     }
 }
+
+pub(crate) struct ErrCallback(Option<Box<dyn Fn(Error) + Send + Sync + 'static>>);
+impl ErrCallback {
+    pub fn call(&self, client: &Client, err: Error) {
+        if let Some(callback) = self.0.as_ref() {
+            callback(err);
+        } else {
+            let si = client.server_info();
+            eprintln!("{} on connection [{}]", err, si.client_id);
```

Comment:
Maybe we should call into `log`(which is configurable) instead of printing directly to stderr here.

Reply:
This is a default version, we could call into log though if we already have it in the client..

Reply:
So with log.error! default does not print anything and I want the default handler, meaning if you do not set one, to print to stderr when bad things happen.